### PR TITLE
feat: add `RingChannel.recv_deadline`

### PR DESCRIPTION
**Pull request by wyfo**
It's needed for Python binding. In fact, `recv` cannot be called in Python because it prevents Python signals, e.g. `KeyboardInterrupt` to be handled. The solution is to call `recv_deadline` within in a loop with a short deadline, e.g. 100ms.

### CODE REVIEW

**Comment by wyfo** [comment]
@Mallets

**Comment by Mallets** [comment]
From a user point of view, should we add also a `recv_timeout(duration: Duration)`? It will simply call `recv_deadline()` under the hood.

File: zenoh/src/api/handlers/ring.rs [hunk 1]
```diff
@@ -62,7 +67,29 @@ impl<T> RingChannelHandler<T> {
...
+    /// Receive from the ring channel with a deadline.
+    ///
+    /// If the ring channel is empty, this call will block until an element is available in the channel,
+    /// or return `None` if the deadline has passed.
+    pub fn recv_deadline(&self, deadline: Instant) -> ZResult<Option<T>> {
+        let Some(channel) = self.ring.upgrade() else {
+            bail!("The ringbuffer has been deleted.");
+        };
+
+        loop {
```
Comment (Mallets) [inline zenoh/src/api/handlers/ring.rs:79]: Why the loop?

Comment (wyfo) [reply zenoh/src/api/handlers/ring.rs:79]: Because of half-backed PR 😓  it should be better now.

Comment (wyfo) [reply zenoh/src/api/handlers/ring.rs:79]: I mean, there is a loop because there is one in `recv`, but I forgot to catch the timeout error.

**Pull request merged**